web/client/components/manager/rulesmanager/ActiveRuleModal.jsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2016, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+const {Modal, Button, Alert} = require('react-bootstrap');
+
+const Message = require('../../I18N/Message');
+const RuleAttributes = require('./RuleAttributes');
+
+const ActiveRuleModal = React.createClass({
+    propTypes: {
+        updateActiveRule: React.PropTypes.func,
+        addRule: React.PropTypes.func,
+        updateRule: React.PropTypes.func,
+        loadRoles: React.PropTypes.func,
+        loadUsers: React.PropTypes.func,
+        loadWorkspaces: React.PropTypes.func,
+        loadLayers: React.PropTypes.func,
+        options: React.PropTypes.object,
+        services: React.PropTypes.object,
+        activeRule: React.PropTypes.object,
+        error: React.PropTypes.object
+    },
+    getDefaultProps() {
+        return {
+            updateActiveRule: () => {},
+            addRule: () => {},
+            updateRule: () => {},
+            activeRule: {},
+            error: {}
+        };
+    },
+    onClose() {
+        this.props.updateActiveRule({}, undefined, false);
+    },
+    getOnSubmitHandler(status) {
+        if (status === "new") {
+            return this.props.addRule;
+        } else if (status === "edit") {
+            return this.props.updateRule;
+        }
+    },
+    getUpdateRuleAttributesHandler() {
+        const status = this.props.activeRule.status;
+        return function(updatedAttributes) {
+            this.props.updateActiveRule(updatedAttributes, status, true);
+        }.bind(this);
+    },
+    render() {
+        const status = this.props.activeRule.status;
+        const showModal = status === "edit" || status === "new";
+        if (!showModal) {
+            return null;
+        }
+        const titleMsgId = "rulesmanager." + status + "Modal";
+        const buttonMsgId = "rulesmanager." + status + "Button";
+        return (
+            <Modal show={showModal} {...this.props} bsSize="small">
```

Comment:
This will make the tests fail randomly. You need to have a modalOptions on the component that you will pass to the Modal, so that in tests you can set animation: false. Look here for an example: https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/data/identify/__tests__/GeocodeViewer-test.jsx